chainer/backends/cuda.py

```diff
@@ -793,3 +793,17 @@ def get_cudnn_dropout_states_core(thread_id):
 
     seed += numpy.uint64(states_id)
     return cudnn.DropoutStates(None, seed)
+
+
+def _get_cudnn_tensor_layout_x(x_layout):
+    if x_layout == chainer.memory_layouts.CUDNN_CHANNEL_FIRST_X:
+        return cuda.cudnn.CUDNN_TENSOR_NCHW
+    assert x_layout == chainer.memory_layouts.CUDNN_CHANNEL_LAST_X
+    return cuda.cudnn.CUDNN_TENSOR_NHWC
+
+
+def _get_cudnn_tensor_layout_w(w_layout):
+    if w_layout == chainer.memory_layouts.CUDNN_CHANNEL_FIRST_W:
```

Comment:
Is there any specific reason why these two functions are separated?

Reply:
cuDNN APIs accept `d_layout` and `w_layout` separately, and in my intuition it's just a "coincidence" that values of these two sets of arguments are identical in cuDNN.

Reply:
Ok, It is safer to leave them as it is now then. It is not a guaranteed behavior